extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+/**
+ * Discussion user log page
+ */
+class SpecialDiscussionsLogController extends WikiaSpecialPageController {
+	// @todo
+	//1. Pagination
+	//2. Add date range input field
+
+	const PAGINATION_SIZE = 50;
+	const DAYS_RANGE = 15;
+	const HTTP_STATUS_OK = 200;
+	const NO_USER_MATCH_ERROR = "Provided username did not match any user";
+
+	static $inputFormTemplate = 'extensions/wikia/SpecialDiscussionsLog/templates/SpecialDiscussionsLog_inputForm.mustache';
+	static $userLogTemplate = 'extensions/wikia/SpecialDiscussionsLog/templates/SpecialDiscussionsLog_userLog.mustache';
+
+	public function __construct() {
+		parent::__construct( 'DiscussionsLog' );
+	}
+
+	public function execute( $subpage ) {
+
+		if ( !$this->checkAccess() ) {
+			throw new \PermissionsError('specialdiscussionslog');
+		}
+
+		$this->setHeaders();
+
+		$this->wg->Out->clearHTML();
+		$this->wg->Out->setPageTitle( wfMessage( 'discussionslog-pagetitle' )->escaped() );
+
+		$output = $this->getInputForm();
+
+		$userName = null;
+		if ( $this->wg->request->wasPosted() ) {
+			$userName = $this->wg->request->getVal( 'username', null );
+			$output .= $this->getUserLog( $userName );
+		}
+
+		$this->wg->Out->addHtml( $output );
+	}
+
+	private function getInputForm() {
+		return \MustacheService::getInstance()->render(
```

Comment:
Maybe we could use default way rendering templates + `sendRequest`?
it's much easier to follow while all controllers use this pattern.